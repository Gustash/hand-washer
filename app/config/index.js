import local from './local'
import development from './development'
import production from './production'
import colors from './colors'
import store from './store'

let config = {
  colors,
  storybookEnabled: false,
}

if (__DEV__) {
  config = {
    ...config,
    ...development,
    ...local,
    ...store,
  }
} else {
  config = {
    ...config,
    ...local,
    ...production,
    ...store,
  }
}

export { colors }
export default config
