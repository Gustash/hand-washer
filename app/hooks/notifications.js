import { useEffect } from 'react'
import { Platform } from 'react-native'
import { useSelector } from 'react-redux'
import PushNotificationIOS from '@react-native-community/push-notification-ios'
import PushNotification from 'react-native-push-notification'

const onNotification = (notification, onNotificationOpen) => {
  const { id } = Platform.select({
    ios: notification.data,
    android: notification,
  })

  // process the notification
  PushNotification.setApplicationIconBadgeNumber(0)

  if (id === '1') {
    // Wash hands notification
    onNotificationOpen()
  }

  // (required) Called when a remote is received or opened, or local notification is opened
  notification.finish(PushNotificationIOS.FetchResult.NoData)
}

/**
 * Self explanatory. Should only be used for unit testing.
 * @type {{onNotification: onNotification}}
 */
export const exportsForTesting = {
  onNotification,
}

export const useScheduledNotification = onNotificationOpen => {
  const shiftActive = useSelector(state => state.shifts.shiftActive)
  const shiftReminders = useSelector(state => state.shifts.shiftReminders)

  useEffect(() => {
    PushNotification.configure({
      onNotification: notification => {
        onNotification(notification, onNotificationOpen)
      },
      permissions: {
        alert: true,
        badge: true,
        sound: true,
      },
      popInitialNotification: true,
      /**
       * (optional) default: true
       * - Specified if permissions (ios) and token (android and ios) will requested or not,
       * - if not, you must call PushNotificationsHandler.requestPermissions() later
       * - if you are not using remote notification or do not have Firebase installed, use this:
       *     requestPermissions: Platform.OS === 'ios'
       */
      requestPermissions: Platform.OS === 'ios',
    })
  }, [onNotificationOpen])

  useEffect(() => {
    if (shiftActive && shiftReminders) {
      PushNotification.localNotificationSchedule({
        /* Android Only Properties */
        id: '1', // (optional) Valid unique 32 bit integer specified as string. default: Autogenerated Unique ID
        ticker: 'Wash your hands!', // (optional)
        autoCancel: true, // (optional) default: true
        largeIcon: 'ic_launcher', // (optional) default: "ic_launcher"
        smallIcon: 'ic_notification', // (optional) default: "ic_notification" with fallback for "ic_launcher"
        priority: 'high', // (optional) set notification priority, default: high
        importance: 'high', // (optional) set notification importance, default: high
        allowWhileIdle: true, // (optional) set notification to work while on doze, default: false
        ignoreInForeground: true, // (optional) if true, the notification will not be visible when the app is in the foreground (useful for parity with how iOS notifications appear)

        /* iOS only properties */
        userInfo: {
          id: '1',
        }, // (optional) default: {} (using null throws a JSON value '<null>' error)

        /* iOS and Android properties */
        title: 'Wash your hands!', // (optional)
        message: "It's been 2 hours since you last washed your hands on shift.", // (required)
        number: 1, // (optional) Valid 32 bit integer specified as string. default: none (Cannot be zero)
        repeatType: Platform.select({
          ios: 'hour', // react-native-push-notification-ios doesn't support repeated notification at specific times, so this is the best we can do
          android: 'time',
        }), // (optional) Repeating interval. Check 'Repeating Notifications' section for more info.
        repeatTime: 7200000, // 2 hours
        date: new Date(Date.now() + 7200000), // 2 hours from now
      })
    } else {
      PushNotification.cancelLocalNotifications({ id: '1' })
    }
  }, [shiftActive, shiftReminders])
}
