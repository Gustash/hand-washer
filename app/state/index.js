export { actions } from './slices'
export { default as rootReducer } from './reducer'
