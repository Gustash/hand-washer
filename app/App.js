import 'react-native-gesture-handler'
import React from 'react'
import { StyleSheet, StatusBar } from 'react-native'
import { Provider } from 'react-redux'
import { PersistGate } from 'redux-persist/integration/react'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5'

import Home from './screens/Home'
import WashTimer from './screens/WashTimer'
import Settings from './screens/Settings'
import config, { colors } from './config'

const Stack = createStackNavigator()

const { store, persistor } = config.createStore()

const App = () => (
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <NavigationContainer>
        <StatusBar backgroundColor={colors.green} />
        <Stack.Navigator
          screenOptions={{
            headerStyle: styles.header,
            headerBackImage: () => (
              <FontAwesome5
                name="chevron-left"
                size={16}
                color={colors.white}
                style={styles.headerBackIcon}
              />
            ),
            headerTitleStyle: styles.headerTitle,
            headerRightContainerStyle: styles.headerRightIcon,
            headerBackTitle: 'Back',
            headerBackTitleStyle: [styles.headerTitle, styles.headerBackTitle],
          }}
        >
          <Stack.Screen
            name="Home"
            component={Home}
            options={{ title: 'Wash Timer' }}
          />
          <Stack.Screen name="Settings" component={Settings} />
          <Stack.Screen name="Timer" component={WashTimer} />
        </Stack.Navigator>
      </NavigationContainer>
    </PersistGate>
  </Provider>
)

const styles = StyleSheet.create({
  header: {
    backgroundColor: colors.green,
  },
  headerTitle: {
    color: colors.white,
    fontSize: 17,
  },
  headerRightIcon: {
    marginRight: 16,
  },
  headerBackTitle: {
    fontSize: 16,
  },
  headerBackIcon: {
    marginLeft: 16,
    marginRight: 4,
  },
})

export default App
