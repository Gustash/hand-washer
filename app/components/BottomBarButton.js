import React from 'react'
import { TouchableOpacity, StyleSheet, Text, SafeAreaView } from 'react-native'
import { colors } from 'app/config'

const BottomBarButton = ({
  onPress,
  text,
  style,
  color = colors.lightBlue,
  textColor = colors.white,
}) => (
  <TouchableOpacity onPress={onPress}>
    <SafeAreaView style={[styles.container, { backgroundColor: color }, style]}>
      <Text style={[styles.text, { color: textColor }]}>{text}</Text>
    </SafeAreaView>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    shadowColor: colors.black,
    shadowOpacity: 0.2,
    shadowOffset: {
      height: -4,
    },
    shadowRadius: 16,
  },
  text: {
    fontSize: 17,
    textAlign: 'center',
    marginVertical: 16,
  },
})

export default BottomBarButton
