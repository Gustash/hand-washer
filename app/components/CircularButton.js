import React from 'react'
import { TouchableOpacity, View, StyleSheet, Text } from 'react-native'
import { colors } from 'app/config'

export const DEFAULT_CIRCULAR_BUTTON_SIZE = 204

const CircularButton = ({
  size = DEFAULT_CIRCULAR_BUTTON_SIZE,
  onPress,
  text,
  icon,
  style,
}) => (
  <TouchableOpacity
    style={[
      styles.container,
      { width: size, height: size, borderRadius: size / 2 },
      style,
    ]}
    onPress={onPress}
    disabled={!onPress}
  >
    <View style={styles.icon}>{icon}</View>
    <Text style={styles.text}>{text}</Text>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.lightBlue,
    shadowColor: colors.black,
    shadowOpacity: 0.2,
    shadowOffset: {
      height: 2,
    },
    shadowRadius: 16,
    padding: 30,

    elevation: 16,
  },
  icon: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: colors.white,
    textAlign: 'center',
    fontSize: 21,
  },
})

export default CircularButton
