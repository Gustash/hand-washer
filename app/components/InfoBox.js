import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { colors } from 'app/config'

const InfoBox = ({ text, style, hidden = false }) => (
  <View style={[styles.container, { opacity: hidden ? 0 : 1 }, style]}>
    <Text style={styles.text}>{text}</Text>
  </View>
)

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.green,
    paddingHorizontal: 22,
    paddingVertical: 6,
    borderRadius: 4,
  },
  text: {
    color: colors.white,
    textAlign: 'center',
    fontSize: 17,
  },
})

export default InfoBox
