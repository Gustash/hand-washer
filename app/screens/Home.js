import React, { useLayoutEffect } from 'react'
import {
  StyleSheet,
  View,
  FlatList,
  Text,
  TouchableOpacity,
} from 'react-native'
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5'
import { useSelector, useDispatch } from 'react-redux'
import moment from 'moment'

import { colors } from 'app/config'
import { actions } from 'app/state'
import InfoBox from 'app/components/InfoBox'
import CircularButton from 'app/components/CircularButton'
import HandsWashIcon from 'app/components/HandsWashIcon'
import BottomBarButton from 'app/components/BottomBarButton'
import LogCard from 'app/components/LogCard'
import { useScheduledNotification } from 'app/hooks/notifications'

const Home = ({ navigation }) => {
  useScheduledNotification(() => {
    navigation.navigate('Timer')
  })

  const dispatch = useDispatch()
  const shiftActive = useSelector(state => state.shifts.shiftActive)
  const shiftReminders = useSelector(state => state.shifts.shiftReminders)
  const washes = useSelector(state => state.history.washes)

  const sortedWashes = [...washes].sort((a, b) =>
    moment(b.date, 'DD-MM-YYYY HH:mm').diff(moment(a.date, 'DD-MM-YYYY HH:mm'))
  )

  useLayoutEffect(() => {
    navigation.setOptions({
      headerRight: () => (
        <TouchableOpacity
          onPress={() => {
            navigation.navigate('Settings')
          }}
        >
          <FontAwesome5 name="cog" size={21} color={colors.white} />
        </TouchableOpacity>
      ),
    })
  }, [navigation])

  return (
    <View style={styles.container}>
      <FlatList
        contentContainerStyle={styles.flatList}
        ListHeaderComponent={
          <>
            <InfoBox
              text={`Currently on shift ${
                shiftReminders
                  ? 'and receiving reminders'
                  : 'with reminders disabled'
              }`}
              style={styles.infoBox}
              hidden={!shiftActive}
            />
            <CircularButton
              text="Wash"
              icon={<HandsWashIcon color={colors.white} size={68} />}
              style={styles.circularButton}
              onPress={() => {
                navigation.navigate('Timer')
              }}
            />
            {washes.length > 0 && (
              <Text testId="home-history-subheader" style={styles.subHeader}>
                HISTORY
              </Text>
            )}
          </>
        }
        data={sortedWashes}
        keyExtractor={({ id }) => `log_${id}`}
        renderItem={({ item }) => (
          <LogCard style={styles.card} date={item.date} />
        )}
      />
      <BottomBarButton
        text={`${shiftActive ? 'End' : 'Start'} Shift`}
        color={shiftActive ? colors.lightBlue : colors.green}
        onPress={() => {
          dispatch(actions.setShiftActive(!shiftActive))
        }}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  flatList: {
    flexGrow: 1,
  },
  infoBox: {
    marginTop: 28,
    marginHorizontal: 24,
  },
  circularButton: {
    alignSelf: 'center',
    marginTop: 20,
  },
  card: {
    marginLeft: 16,
  },
  subHeader: {
    fontSize: 14,
    color: colors.darkGrey,
    marginVertical: 28,
    marginHorizontal: 32,
  },
})

export default Home
