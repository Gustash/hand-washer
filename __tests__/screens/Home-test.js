import React from 'react'
import { Text, FlatList, TouchableOpacity } from 'react-native'
import { Provider, useDispatch, useSelector } from 'react-redux'
import configureStore from 'redux-mock-store'
import TestRenderer from 'react-test-renderer'
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5'

import Home from '../../app/screens/Home'
import CircularButton from '../../app/components/CircularButton'
import BottomBarButton from '../../app/components/BottomBarButton'
import { useScheduledNotification } from '../../app/hooks/notifications'
import { actions } from '../../app/state'
import InfoBox from '../../app/components/InfoBox'

const mockStore = configureStore([])

const mockDispatch = jest.fn()

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}))

jest.mock('app/hooks/notifications', () => ({
  ...jest.requireActual('app/hooks/notifications'),
  useScheduledNotification: jest.fn(),
}))

describe('<Home />', () => {
  let store
  let wrapper
  let currentOptions

  const renderWithProvider = () =>
    TestRenderer.create(
      <Provider store={store}>
        <Home
          navigation={{
            navigate: jest.fn(),
            setOptions: jest.fn(options => {
              currentOptions = options
            }),
          }}
        />
      </Provider>
    )

  beforeEach(() => {
    mockDispatch.mockReset()
    useScheduledNotification.mockReset()

    currentOptions = {}
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [
          { id: 1, date: '01-01-1990 00:00' },
          { id: 2, date: '02-01-1990 01:00' },
          { id: 3, date: '01-01-1990 23:00' },
        ],
      },
      shifts: {
        shiftActive: false,
        shiftReminders: false,
      },
    })

    useSelector.mockImplementation(selector => selector(store.getState()))
    useDispatch.mockImplementation(() => mockDispatch)

    TestRenderer.act(() => {
      wrapper = renderWithProvider(store)
    })
  })

  it('matches snapshot', () => {
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('creates right header icon', () => {
    expect(
      wrapper.root.findByType(Home).props.navigation.setOptions
    ).toHaveBeenCalledTimes(1)
    expect(currentOptions.headerRight()).toEqual(
      <TouchableOpacity onPress={expect.any(Function)}>
        <FontAwesome5 name="cog" size={21} color="#FFFFFF" />
      </TouchableOpacity>
    )
  })

  it('navigates to Timer on notification open', () => {
    useScheduledNotification.mockReset()
    // Instantly call callback to mock the user opening a notification on the device
    useScheduledNotification.mockImplementation(cb => cb())

    TestRenderer.act(() => {
      wrapper = renderWithProvider(store)
    })

    expect(useScheduledNotification).toHaveBeenCalledTimes(1)

    expect(
      wrapper.root.findByType(Home).props.navigation.navigate
    ).toHaveBeenCalledWith('Timer')
  })

  it('sorts data by date descending', () => {
    const expectedData = [
      { id: 2, date: '02-01-1990 01:00' },
      { id: 3, date: '01-01-1990 23:00' },
      { id: 1, date: '01-01-1990 00:00' },
    ]

    expect(wrapper.root.findByType(FlatList).props.data).toEqual(expectedData)
  })

  it('navigates to Settings when header right icon is pressed', () => {
    const headerRight = TestRenderer.create(currentOptions.headerRight())

    headerRight.root.findByType(TouchableOpacity).props.onPress()

    const { navigate } = wrapper.root.findByType(Home).props.navigation

    expect(navigate).toHaveBeenCalledTimes(1)
    expect(navigate).toHaveBeenCalledWith('Settings')
  })

  it('should navigate to Timer when CircleButton is pressed', () => {
    wrapper.root.findByType(CircularButton).props.onPress()

    const { navigate } = wrapper.root.findByType(Home).props.navigation

    expect(navigate).toHaveBeenCalledTimes(1)
    expect(navigate).toHaveBeenCalledWith('Timer')
  })

  it('should enable shift active when BottomBarButton is pressed', () => {
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [],
      },
      shifts: {
        shiftActive: false,
        shiftReminders: false,
      },
    })

    wrapper = renderWithProvider()

    const bottomBarButton = wrapper.root.findByType(BottomBarButton)
    const { onPress } = bottomBarButton.props

    onPress()

    expect(mockDispatch).toHaveBeenCalledTimes(1)
    expect(mockDispatch).toHaveBeenCalledWith(actions.setShiftActive(true))
  })

  it('should disable shift active when BottomBarButton is pressed', () => {
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [],
      },
      shifts: {
        shiftActive: true,
        shiftReminders: false,
      },
    })

    wrapper = renderWithProvider()

    const bottomBarButton = wrapper.root.findByType(BottomBarButton)
    const { onPress } = bottomBarButton.props

    onPress()

    expect(mockDispatch).toHaveBeenCalledTimes(1)
    expect(mockDispatch).toHaveBeenCalledWith(actions.setShiftActive(false))
  })

  it('extracts keys from data', () => {
    const flatList = wrapper.root.findByType(FlatList)

    expect(
      flatList.props.keyExtractor({ id: 1, date: '01-01-1990 00:00' })
    ).toBe('log_1')
  })

  it('hides info box when not on shift', () => {
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [],
      },
      shifts: {
        shiftActive: false,
        shiftReminders: false,
      },
    })

    wrapper = renderWithProvider()

    expect(wrapper.root.findByType(InfoBox).props.hidden).toBeTruthy()
  })

  it('shows info box when on shift', () => {
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [],
      },
      shifts: {
        shiftActive: true,
        shiftReminders: false,
      },
    })

    wrapper = renderWithProvider()

    expect(wrapper.root.findByType(InfoBox).props.hidden).toBeFalsy()
  })

  it('warns about receiving reminders', () => {
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [],
      },
      shifts: {
        shiftActive: true,
        shiftReminders: true,
      },
    })

    wrapper = renderWithProvider()

    const expectedText = 'Currently on shift and receiving reminders'

    expect(wrapper.root.findByType(InfoBox).props.text).toEqual(expectedText)
  })

  it('warns about not receiving reminders', () => {
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [],
      },
      shifts: {
        shiftActive: true,
        shiftReminders: false,
      },
    })

    wrapper = renderWithProvider()

    const expectedText = 'Currently on shift with reminders disabled'

    expect(wrapper.root.findByType(InfoBox).props.text).toEqual(expectedText)
  })

  it('renders history subheader when there are washes', () => {
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [{ id: 1, date: '01-01-1990 00:00' }],
      },
      shifts: {
        shiftActive: true,
        shiftReminders: false,
      },
    })

    wrapper = renderWithProvider()

    const subheader = wrapper.root
      .findAllByType(Text)
      .find(({ props }) => props.testId === 'home-history-subheader')

    expect(subheader).toBeDefined()
  })

  it('hides history subheader when there are no washes', () => {
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [],
      },
      shifts: {
        shiftActive: true,
        shiftReminders: false,
      },
    })

    wrapper = renderWithProvider()

    const subheader = wrapper.root
      .findAllByType(Text)
      .find(({ props }) => props.testId === 'home-history-subheader')

    expect(subheader).not.toBeDefined()
  })

  it('renders start shift button', () => {
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [],
      },
      shifts: {
        shiftActive: false,
        shiftReminders: false,
      },
    })

    wrapper = renderWithProvider()

    expect(wrapper.root.findByType(BottomBarButton).props).toMatchObject({
      text: 'Start Shift',
      color: '#00A778',
      onPress: expect.any(Function),
    })
  })

  it('renders end shift button', () => {
    store = mockStore({
      history: {
        storeWashes: true,
        washes: [],
      },
      shifts: {
        shiftActive: true,
        shiftReminders: false,
      },
    })

    wrapper = renderWithProvider()

    expect(wrapper.root.findByType(BottomBarButton).props).toMatchObject({
      text: 'End Shift',
      color: '#00A1D2',
      onPress: expect.any(Function),
    })
  })
})
