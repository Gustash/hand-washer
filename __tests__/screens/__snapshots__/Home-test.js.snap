// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={
      <React.Fragment>
        <InfoBox
          hidden={true}
          style={
            Object {
              "marginHorizontal": 24,
              "marginTop": 28,
            }
          }
          text="Currently on shift with reminders disabled"
        />
        <CircularButton
          icon={
            <HandsWashIcon
              color="#FFFFFF"
              size={68}
            />
          }
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "marginTop": 20,
            }
          }
          text="Wash"
        />
        <Text
          style={
            Object {
              "color": "#4A4A4A",
              "fontSize": 14,
              "marginHorizontal": 32,
              "marginVertical": 28,
            }
          }
          testId="home-history-subheader"
        >
          HISTORY
        </Text>
      </React.Fragment>
    }
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    data={
      Array [
        Object {
          "date": "02-01-1990 01:00",
          "id": 2,
        },
        Object {
          "date": "01-01-1990 23:00",
          "id": 3,
        },
        Object {
          "date": "01-01-1990 00:00",
          "id": 1,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#00A778",
                "borderRadius": 4,
                "paddingHorizontal": 22,
                "paddingVertical": 6,
              },
              Object {
                "opacity": 0,
              },
              Object {
                "marginHorizontal": 24,
                "marginTop": 28,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 17,
                "textAlign": "center",
              }
            }
          >
            Currently on shift with reminders disabled
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#00A1D2",
              "borderRadius": 102,
              "elevation": 16,
              "height": 204,
              "marginTop": 20,
              "opacity": 1,
              "padding": 30,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 2,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 16,
              "width": 204,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              aria-hidden="true"
              bbHeight={68}
              bbWidth={68}
              className="prefix__svg-inline--fa prefix__fa-hands-wash prefix__fa-w-18"
              data-icon="hands-wash"
              data-prefix="fas"
              focusable={false}
              height={68}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 68,
                    "width": 68,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={576}
              width={68}
            >
              <RNSVGGroup>
                <RNSVGPath
                  d="M496 224a48 48 0 10-48-48 48 48 0 0048 48zm-184.53-45.55A56.77 56.77 0 01328 176a56 56 0 0119 3.49l15.35-48.61A24 24 0 00342 99.74c-11.53-1.35-22.21 6.44-25.71 17.51l-20.9 66.17zM93.65 386.33c.8-.19 1.54-.54 2.35-.71v-25.69a156 156 0 01107.06-148l73.7-22.76 34.16-108.12a24 24 0 00-20.33-31.11c-11.53-1.34-22.22 6.45-25.72 17.52l-33.45 106.42a8 8 0 01-15.26-4.83l43.37-137.79A24 24 0 00239.2.15C227.67-1.19 217 6.6 213.49 17.66l-47.93 151.71a8 8 0 11-15.26-4.82l38.56-122a24 24 0 00-20.33-31.11C157 10 146.32 17.83 142.82 28.9l-60 189.85-2.06-50.05a24 24 0 00-23.86-24.15c-13.23-.05-24.72 10.54-24.9 23.86v112.73a123.69 123.69 0 0061.65 105.19zM519.1 336H360a8 8 0 010-16h128a24 24 0 0023.54-28.76c-2.19-11.4-12.83-19.24-24.44-19.24H288l47.09-17.06a24 24 0 00-14.18-45.88l-107.72 33.25A123.88 123.88 0 00128 360v25.65a79.78 79.78 0 0158 108.63A118.9 118.9 0 00248 512h208a24 24 0 0023.54-28.76c-2.19-11.4-12.83-19.24-24.44-19.24H360a8 8 0 010-16h128a24 24 0 0023.54-28.76c-2.19-11.4-12.83-19.24-24.44-19.24H360a8 8 0 010-16h160a24 24 0 0023.54-28.76c-2.19-11.4-12.83-19.24-24.44-19.24zM416 64a32 32 0 10-32-32 32 32 0 0032 32zM112 416a48 48 0 1048 48 48 48 0 00-48-48z"
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 21,
                "textAlign": "center",
              }
            }
          >
            Wash
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#4A4A4A",
              "fontSize": 14,
              "marginHorizontal": 32,
              "marginVertical": 28,
            }
          }
          testId="home-history-subheader"
        >
          HISTORY
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "paddingVertical": 16,
              },
              Object {
                "borderBottomColor": "#E6E6E6",
                "borderBottomWidth": 1,
              },
              Object {
                "marginLeft": 16,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#00A778",
                  "justifyContent": "center",
                  "marginHorizontal": 8,
                  "paddingHorizontal": 14,
                  "paddingVertical": 8,
                  "width": 80,
                },
                undefined,
              ]
            }
            testId="row-highlight-container"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 17,
                  "textAlign": "center",
                }
              }
              testId="row-highlight-text"
            >
              01:00
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#4A4A4A",
                  "fontSize": 17,
                }
              }
              testId="row-primary-text"
            >
              Tuesday
            </Text>
            <Text
              style={
                Object {
                  "color": "#9B9B9B",
                  "fontSize": 14,
                }
              }
              testId="row-secondary-text"
            >
              2nd January 1990
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "paddingVertical": 16,
              },
              Object {
                "borderBottomColor": "#E6E6E6",
                "borderBottomWidth": 1,
              },
              Object {
                "marginLeft": 16,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#00A778",
                  "justifyContent": "center",
                  "marginHorizontal": 8,
                  "paddingHorizontal": 14,
                  "paddingVertical": 8,
                  "width": 80,
                },
                undefined,
              ]
            }
            testId="row-highlight-container"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 17,
                  "textAlign": "center",
                }
              }
              testId="row-highlight-text"
            >
              23:00
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#4A4A4A",
                  "fontSize": 17,
                }
              }
              testId="row-primary-text"
            >
              Monday
            </Text>
            <Text
              style={
                Object {
                  "color": "#9B9B9B",
                  "fontSize": 14,
                }
              }
              testId="row-secondary-text"
            >
              1st January 1990
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "paddingVertical": 16,
              },
              Object {
                "borderBottomColor": "#E6E6E6",
                "borderBottomWidth": 1,
              },
              Object {
                "marginLeft": 16,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#00A778",
                  "justifyContent": "center",
                  "marginHorizontal": 8,
                  "paddingHorizontal": 14,
                  "paddingVertical": 8,
                  "width": 80,
                },
                undefined,
              ]
            }
            testId="row-highlight-container"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 17,
                  "textAlign": "center",
                }
              }
              testId="row-highlight-text"
            >
              00:00
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#4A4A4A",
                  "fontSize": 17,
                }
              }
              testId="row-primary-text"
            >
              Monday
            </Text>
            <Text
              style={
                Object {
                  "color": "#9B9B9B",
                  "fontSize": 14,
                }
              }
              testId="row-secondary-text"
            >
              1st January 1990
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": -4,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 16,
          },
          Object {
            "backgroundColor": "#00A778",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 17,
              "marginVertical": 16,
              "textAlign": "center",
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        Start Shift
      </Text>
    </RCTSafeAreaView>
  </View>
</View>
`;
